### ISSUE TITLE
JS_SYNTAX_ERROR from seemingly valid code

### ISSUE THREAD

**Issue by dr-off** [mozilla/addons-linter#3673]
### Describe the problem and steps to reproduce it:
I am trying to submit an update for my web extension but the validation fails due to this code:

```js
class Setting
{
	static instances = [];
	static instancesById = {};

	static categories = new Map();

       // Irrelevant code truncated, the full file is linked at the end of this issue.
}
```

You can reproduce it by including this code in a web extension.

### What happened?
A `JS_SYNTAX_ERROR` is shown:

```
JavaScript syntax error (Parsing as module error:
Unexpected token = at line: 3 and column: 19)
(Parsing as script error: Unexpected token = at
line: 3 and column: 19)
```

This code works fine in Firefox and Chrome and seems valid by all accounts I can find but for some reason, the linter considers it invalid.

### What did you expect to happen?
I would expect it to consider this valid.

### Anything else we should know?
The full code of the class can be found in my repo [here](https://github.com/dr-off/ao3-enhancement-suite/blob/master/js/classes/Setting.js).

**Comment by willdurand** [comment]
> This code works fine in Firefox and Chrome and seems valid by all accounts I can find but for some reason, the linter considers it invalid.

Hi, is this the entire message? You should have a message about unsupported syntax, which is valid but not supported yet because not "fully standardized".

**Comment by dr-off** [comment]
My apologies, I should've included the description as well.

```
There is a JavaScript syntax error in your code, which might be related to some experimental
JavaScript features that aren't an official part of the language specification and therefore not
supported yet. The validation cannot continue on this file.
```

Is assigning static variables in this way not a standardized feature?

**Comment by willdurand** [comment]
Please see: https://github.com/mozilla/addons-linter/issues/3062

**Comment by dr-off** [comment]
Hmm, I see. So I basically just have to work around this for now then?

**Comment by willdurand** [comment]
> Hmm, I see. So I basically just have to work around this for now then?

Yes, unfortunately we don't have a good workaround for that.

**Comment by erosman** [comment]
> So I basically just have to work around this for now then?

ATM, you have to define them outside the `class`.


> [Instance properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#instance_properties)
> 
> Static (class-side) data properties and prototype data properties must be defined outside of the ClassBody declaration:

**Comment by wagnerand** [comment]
Closing as duplicate of #3062.
